cells/lib/ops/mkStandardOCI.nix

```diff
@@ -68,20 +68,82 @@ in
       then debugOperable
       else operable;
 
-    setupLinks = cell.ops.mkSetup "links" [] ''
-      mkdir -p $out/bin
-      ${runtimeEntryLink}
-      ${debugEntryLink}
-      ${livenessLink}
-      ${readinessLink}
+    setupLinks =
+      cell.ops.mkSetup "links" [
+        {
+          regex = "/bin";
+          mode = "0555";
+        }
+      ] ''
+        mkdir -p $out/bin
+        ${runtimeEntryLink}
+        ${debugEntryLink}
+        ${livenessLink}
+        ${readinessLink}
+      '';
+
+    setupCerts = cell.ops.mkSetup "ssl" [] ''
+      mkdir -p $out/etc
+      cp -r ${nixpkgs.cacert}/* $out
+    '';
+
+    setupEtcHack = cell.ops.mkSetup "etc" [] ''
+      mkdir -p $out/etc
+      cat << REASON > $out/etc/.ensure-etc-is-no-symlink
+        # docker creates some minimal files in /etc at runtime
+        # that would otherwise shadow a top-level symlink:
+        #   bash-5.2$ ls /etc
+        #   hostname  hosts  mtab  resolv.conf
+        # therefore, we make trick nixpks.buildenv into not
+        # generating symlinks at the level of /etc but at least
+        # one level deeper by artifie of this overlapping element
+        # that buildenv will detect and push the linking one level down
+      REASON
+    '';
+
+    setupSparseUser = cell.ops.mkUser {
+      user = "nobody";
+      group = "nogroup";
+      inherit uid gid;
+    };
+
+    tmp = nixpkgs.runCommand "tmp" {} ''
+      mkdir -p $out/tmp
     '';
   in
     cell.ops.mkOCI (
       {
-        inherit name uid gid labels options perms config meta;
+        inherit name uid gid labels meta;
         entrypoint = operable';
-        setup = [setupLinks] ++ setup;
+        setup = [setupLinks setupCerts setupEtcHack] ++ setup;
         runtimeInputs = operable.passthru.runtimeInputs;
+        options = options // {copyToRoot = options.copyToRoot or [] ++ [tmp];};
+        config =
+          config
+          // {
+            Env =
+              config.Env
+              or []
+              ++ [
+                # Adopted from: <nixpks>/pkgs/data/misc/cacert/setup-hook.sh
+                # This file is created when nixpkgs.cacert is linked to the root
+                "NIX_SSL_CERT_FILE=/etc/ssl/certs/ca-bundle.crt"
```

Comment:
Is this only for the `nix` bin? We probably don't want that in the container most times.

Reply:
True! Its existence might confuse unnecessarily.

Reply:
I'd be okay with having this as default since it does not take any extra space/build time
What I am wondering, doesn't NIX_SSL_CERT_FILE default to this location?

Reply:
> What I am wondering, doesn't NIX_SSL_CERT_FILE default to this location?

In the newer version of the branch, I removed it and I'm using `dockerTools.caCertificates` instead.

This package explains that, unlike the `cacert` package directly, it copies the certs to the canonical locations  for various distros.

However, it omits to set the env variable for, e.g. the haskell ssl implememtation, which does not implement a canonical location, but requires the explicit hint from the environment (what a good choice!).

For that reason, I left openssl amd Haskell compat, even if the former would have otherwise a canonical detection depending on the distro's build of it.

This ends up a precedence for setting env variables for tools and language systems which implement SSL, so I left it, but without unnecessarily hinting at Nix being a special kid.